Add section on finding errors from polkascan and subscan (#1425)
w3f/polkadot-wiki: docs/maintain-errors.md

If you cannot look up the error this way, or there is no message in the `details` field, consult the
table below.

## Polkascan and Subscan

> Polkascan and Subscan show the `ExtrnsicFailed` event when a transaction does not succeed

Reviewer: ```suggestion
Polkascan and Subscan show the `ExtrinsicFailed` event when a transaction does not succeed
```

([example](https://polkascan.io/polkadot/event/2836233-3)). This event gives us the `error` and
`index` indices of the error but does not give us a nice message to understand what it means. We
will look up the error in the codebase ourselves to understand what went wrong.

First, we should understand that the `index` number is the index of the pallet in the runtime from
which the error originated. The `error` is likewise the index of that pallet's errors which is the
exact one we're looking for. Both of these indices start counting from 0.

> For example, if we have `index` is 5 and `index` is 3 like in the example linked above, we need to

Reviewer: ```suggestion
For example, if `index` is 5 and `error` is 3, as in the example linked above, we need to
```

> look at the runtime for the 4th error in the pallet at index 5.

Reviewer: ```suggestion
look at the runtime for the fourth error (index 3) in the sixth pallet (index 5).
```

By looking at the
[runtime code](https://github.com/paritytech/polkadot/blob/master/runtime/polkadot/src/lib.rs#L914)

> we see that the pallet at index 3 is "Timestamp". Now we will check the timestamp pallet's code

Reviewer: ```suggestion
we see that the pallet at index 5 is "Balances". Now we will check the Balances pallet's code
```

> which is hosted in the Substrate repository. According to

Reviewer: ```suggestion
which is hosted in the Substrate repository, and look for the fourth error in the Error enum. According to
```
